### PR TITLE
[CPU][ARM] Correct execution of the Reorder via Transpose

### PR DESCRIPTION
**Pull request by EgorDuplensky**
### TBD:
- [x] Update commit message

### Tickets:
- 121173

### CODE REVIEW

**Comment by EgorDuplensky** [comment]
@dmitry-gorokhov Could you please take a look

File: src/plugins/intel_cpu/src/nodes/executors/common/ref_transpose.hpp [hunk 1]
```diff
@@ -28,6 +28,10 @@ class RefTransposeExecutorBuilder : public TransposeExecutorBuilder {
     bool isSupported(const TransposeParams& transposeParams,
                      const std::vector<MemoryDescPtr>& srcDescs,
                      const std::vector<MemoryDescPtr>& dstDescs) const override {
+        // only ncsp layout is currently supported
+        if (!srcDescs[0]->hasLayoutType(LayoutType::ncsp))
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/executors/common/ref_transpose.hpp:32]: ref transpose impl looks pretty generic: it [utilizes](https://github.com/openvinotoolkit/openvino/blob/574a642c74689c0fdee63003012bfafbae60df57/src/plugins/intel_cpu/src/nodes/executors/common/ref_transpose.cpp#L59-L60) strides to get correct element offset. Stride itself already [accomodates ](https://github.com/openvinotoolkit/openvino/blob/574a642c74689c0fdee63003012bfafbae60df57/src/plugins/intel_cpu/src/memory_desc/cpu_blocked_memory_desc.cpp#L51-L62) layout details based on order field. In other words I would expect ref impl to fully support non-blocked layouts.

Comment (EgorDuplensky) [reply src/plugins/intel_cpu/src/nodes/executors/common/ref_transpose.hpp]: you're right, the problem was in incorrectly passed transposed parameters.

**Review by dmitry-gorokhov — APPROVED** [review]

**Comment by wkobielx** [comment]
Manually testing postmerge macos arm64 build+tests: https://openvino-ci.toolbox.iotg.sclab.intel.com/job/private-ci/job/ie/job/build-linux-macos_arm64/5270/

**Comment by wkobielx** [comment]
Testing macos arm64 build + tests: https://openvino-ci.toolbox.iotg.sclab.intel.com/job/private-ci/job/ie/job/build-linux-macos_arm64/5273/

**Comment by alvoron** [comment]
> Testing macos arm64 build + tests: https://openvino-ci.toolbox.iotg.sclab.intel.com/job/private-ci/job/ie/job/build-linux-macos_arm64/5273/
> 
> build ok, 10 tests failed

Original failed tests are covered by https://github.com/openvinotoolkit/openvino/pull/20115

**Pull request merged**